Bottom: fdbcecf2b04fa810082fb43523b884b48aa2a599
Top:    d365ff2265ea44aa16d8b00965124c9670e6f726
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-11 16:10:53 +0000

Workaround crash where contour is passed as NULL to poly_PreContour













---

diff --git a/src/polygon.c b/src/polygon.c
index 1999b43..471b91c 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -253,6 +253,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
